include/flox/core/exceptions.hh

```diff
@@ -99,6 +100,9 @@ public:
   //   methods.
   [[nodiscard]] std::string
   what_string() const noexcept;
+
+  [[nodiscard]] nlohmann::json
```

Comment:
@aakropotkin what is the purpose of `[[nodiscard]]`?

Reply:
Good question.
When we return values sometimes the _expected usage_ is for users to assign the return value to a local variable, or otherwise _use the returned value_.
In other cases a routine might really be called for it's side effects, and using the returned value is _optional_. If the user decides not to use the result this is called "discarding".

`[[nodiscard]]` indicates that a returned value should not be discarded. If a user doesn't use the returned value compilers may throw warnings or errors to indicate that the caller is using an interface improperly.

In rare cases that you really want to discard the result of a `[[nodiscard]]` function ( perhaps just for testing to ensure that exceptions aren't thrown ), you can `try { (void) foo.getFoo() } catch( ... ) { ... }` to explicitly discard the result without triggering warnings/errors. This usage is rare but it appears in our test suite so I wanted to highlight it.